.github/workflows/cancel-previous-pr-workflows.yml

```diff
@@ -1,11 +1,14 @@
 name: Cancel Previous PR Workflows
 on:
-  pull_request:
-    branches: [ main ]
+  workflow_run:
+    workflows: ["Dependencies autoupdate", "Build Agents", "Build Anomaly Detection App Container", "Build Controller", "Build ONVIF Broker (.NET)", "Build OPC UA Monitoring Broker", "Build OpenCV Base", "Build Production Rust Code", "Build Rust CrossBuild", "Build UDEV Broker", "Build Video Streaming App Container", "Build Webhook Configuration", "Check Rust", "Check versioning", "Helm", "Tarpaulin Code Coverage", "Test K3s, Kubernetes, and MicroK8s"]
+    types:
+      - requested
 jobs:
   cancel:
     runs-on: ubuntu-latest
+    if: (github.event.workflow_run.event != 'release') && (github.event.workflow_run.event != 'push')
```

Comment:
nit: since this is titled "Cancel Previous **PR** Workflows", you might make this conditional just for PRs ... `if: startsWith(github.event_name, 'pull_request')` or `if: (github.event_name == 'pull_request')`

Reply:
(functionally, my suggestion is probably the same now, but if we added a trigger for some other event like `label added`, this would need to be updated)

Reply:
Great point! Adjusted in acb3083e631870c8. I used exact match instead of `startsWith` since we are avoiding `pull_request_target`.